bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/hbasesnapshots/ImportJobFromHbaseSnapshot.java

```diff
@@ -0,0 +1,139 @@
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.io.hadoop.format.HadoopFormatIO;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+
+/**
+ * A job that imports data from HBase snapshot exports hosted in Cloud Storage bucket into Cloud
+ * Bigtable.
+ *
+ * <p>Example: If you have exported your HBase Snapshot to GCS bucket gs://$HBASE_EXPORT_ROOT_PATH
+ * and want to import snapshot gs://$HBASE_EXPORT_ROOT_PATH/.hbase-snapshot/$SNAPSHOT_NAME into
+ * Cloud Bigtable $TABLE in $INSTANCE, execute the following command to run the job directly:
+ *
+ * <pre>
+ * mvn compile exec:java \
+ *   -DmainClass=com.google.cloud.bigtable.beam.hbasesnapshots.ImportJobFromHbaseSnapshot \
+ *   -Dexec.args="--runner=DataflowRunner \
+ *                --stagingLocation=gs://$STAGING_PATH \
+ *                --project=$PROJECT \
+ *                --bigtableInstanceId=$INSTANCE \
+ *                --bigtableTableId=$TABLE \
+ *                --hbaseSnapshotSourceDir=gs://$HBASE_EXPORT_ROOT_PATH \
+ *                --snapshotName=$SNAPSHOT_NAME  \
+ *                --restoreDir=gs://$RESTORE_PATH
+ * </pre>
+ *
+ * Note: restoreDir is a GCS path used to save temporal files generated by Snapshot scans. and it
+ * must not be a sub-folder of the hbaseSnapshotSourceDir.
+ */
+@InternalExtensionOnly
+public class ImportJobFromHbaseSnapshot {
+  private static final Log LOG = LogFactory.getLog(ImportJobFromHbaseSnapshot.class);
+
+  public interface ImportOptions extends ImportJob.ImportOptions {
+    @Description("The HBase root dir where HBase snapshot files resides.")
+    String getHbaseSnapshotSourceDir();
+
+    @SuppressWarnings("unused")
+    void setHbaseSnapshotSourceDir(String hbaseSnapshotSourceDir);
+
+    @Description("Temporal location for restoring snapshots")
+    String getRestoreDir();
+
+    @SuppressWarnings("unused")
+    void setRestoreDir(String restoreDir);
+
+    @Description("Snapshot name")
+    String getSnapshotName();
+
+    @SuppressWarnings("unused")
+    void setSnapshotName(String snapshotName);
+  }
+
+  public static void main(String[] args) {
+    PipelineOptionsFactory.register(ImportOptions.class);
+
+    ImportOptions opts =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(ImportOptions.class);
+
+    LOG.info("DEBUG===> Building Pipeline");
```

Comment:
`LOG.info("DEBUG`, this looks a bit odd. Why is DEBUG==> there?

Reply:
Same below

Reply:
Removed.